File: test_pagination.dart
import 'lib/services/livescore_api_service.dart';

/// Test per verificare che la paginazione recuperi più di 30 partite
void main() async {
  print('🔍 TEST PAGINAZIONE API LIVESCORE');
  print('=' * 60);
  
  final service = LiveScoreApiService();
  
  try {
    print('\n📋 Recupero partite di oggi con paginazione...\n');
    
    final fixtures = await service.getFixturesToday();
    
    print('\n' + '=' * 60);
    print('✅ RISULTATO FINALE');
    print('=' * 60);
    print('📊 Totale partite recuperate: ${fixtures.length}');
    
    if (fixtures.length > 30) {
      print('✅ SUCCESSO! Recuperate più di 30 partite grazie alla paginazione!');
    } else if (fixtures.length == 30) {
      print('⚠️ Recuperate esattamente 30 partite. Potrebbero esserci più pagine disponibili.');
    } else {
      print('ℹ️ Recuperate ${fixtures.length} partite (meno di 30, probabilmente tutte le disponibili oggi)');
    }
    
    // Analizza le partite per paese
    final Map<String, int> countryCounts = {};
    for (final fixture in fixtures) {
      final country = fixture.country.isNotEmpty ? fixture.country : 'Unknown';
      countryCounts[country] = (countryCounts[country] ?? 0) + 1;
    }
    
    print('\n📊 DISTRIBUZIONE PER PAESE (Top 15):');
    final sortedCountries = countryCounts.entries.toList()
      ..sort((a, b) => b.value.compareTo(a.value));
    for (int i = 0; i < (sortedCountries.length > 15 ? 15 : sortedCountries.length); i++) {
      final entry = sortedCountries[i];
      print('   ${i + 1}. ${entry.key}: ${entry.value} partite');
    }
    
    // Cerca partite internazionali
    final internationalFixtures = fixtures.where((fixture) {
      final league = fixture.league.toLowerCase();
      return league.contains('champions') || 
             league.contains('europa') || 
             league.contains('world') ||
             league.contains('uefa') ||
             league.contains('fifa') ||
             league.contains('international');
    }).toList();
    
    if (internationalFixtures.isNotEmpty) {
      print('\n🌍 PARTITE INTERNAZIONALI TROVATE: ${internationalFixtures.length}');
      for (final fixture in internationalFixtures.take(10)) {
        print('   - ${fixture.home} vs ${fixture.away}');
        print('     ${fixture.league} (${fixture.country})');
      }
    } else {
      print('\n⚠️ NESSUNA PARTITA INTERNAZIONALE OGGI');
      print('   (Champions League e Europa League non si giocano tutti i giorni)');
    }
    
    // Mostra alcune partite di esempio
    print('\n📝 ESEMPI DI PARTITE (prime 10):');
    for (int i = 0; i < (fixtures.length > 10 ? 10 : fixtures.length); i++) {
      final fixture = fixtures[i];
      final timeStr = '${fixture.start.hour.toString().padLeft(2, '0')}:${fixture.start.minute.toString().padLeft(2, '0')}';
      print('   ${i + 1}. ${fixture.home} vs ${fixture.away}');
      print('      ${fixture.league} (${fixture.country})');
      print('      Orario: $timeStr');
    }
    
    print('\n' + '=' * 60);
    print('✅ TEST COMPLETATO CON SUCCESSO');
    print('=' * 60);
    
  } catch (e) {
    print('\n❌ ERRORE: $e');
    print('\nStack trace:');
    print(e);
  }
}

File: test_pagination.js
const axios = require('axios');

console.log('🧪 TEST PAGINAZIONE API LIVESCORE');
console.log('============================================================\n');

async function testPagination() {
  const today = new Date().toISOString().split('T')[0];
  const allMatches = [];
  
  console.log(`📅 Data: ${today}\n`);
  
  // Prova a recuperare più pagine
  for (let page = 1; page <= 5; page++) {
    console.log(`\n📄 Tentativo pagina ${page}...`);
    
    try {
      const response = await axios.get('https://livescore-api.com/api-client/fixtures/matches.json', {
        params: {
          key: 'wUOF0E1DmdetayWk',
          secret: 'Vng53xQ0F9Knz416YPLZuNCR1Rkbqhvl',
          date: today,
          page: page
        },
        timeout: 15000
      });
      
      if (response.status === 200 && response.data) {
        const fixtures = response.data?.data?.fixtures || [];
        console.log(`✅ Pagina ${page}: ${fixtures.length} partite`);
        
        if (fixtures.length === 0) {
          console.log(`⚠️ Nessuna partita trovata, fine paginazione`);
          break;
        }
        
        // Aggiungi le partite uniche
        fixtures.forEach(match => {
          const key = `${match.home_name || match.home?.name}-${match.away_name || match.away?.name}`;
          if (!allMatches.find(m => m.key === key)) {
            allMatches.push({
              key,
              home: match.home_name || match.home?.name,
              away: match.away_name || match.away?.name,
              league: match.competition?.name || match.competition_name,
              page: page
            });
          }
        });
        
        console.log(`📊 Totale partite uniche finora: ${allMatches.length}`);
        
        // Se la pagina ha meno di 30 partite, probabilmente è l'ultima
        if (fixtures.length < 30) {
          console.log(`⚠️ Pagina incompleta (${fixtures.length} < 30), probabilmente ultima pagina`);
          break;
        }
        
      } else {
        console.log(`❌ Pagina ${page}: Status ${response.status}`);
        break;
      }
      
      // Delay tra le richieste
      await new Promise(resolve => setTimeout(resolve, 2000));
      
    } catch (error) {
      console.log(`❌ Errore pagina ${page}: ${error.message}`);
      if (error.response?.status === 404) {
        console.log(`⚠️ Pagina non trovata, fine paginazione`);
        break;
      }
    }
  }
  
  console.log(`\n============================================================`);
  console.log(`📊 RISULTATO FINALE`);
  console.log(`============================================================`);
  console.log(`✅ Totale partite uniche: ${allMatches.length}`);
  
  if (allMatches.length > 0) {
    console.log(`\n🎯 Tutte le partite trovate:\n`);
    allMatches.forEach((match, i) => {
      console.log(`${i + 1}. ${match.home} vs ${match.away}`);
      console.log(`   League: ${match.league} | Pagina: ${match.page}`);
    });
  }
}

testPagination();